### PR TITLE
rework booking app to use GoRouter package for navigation

### PR DESCRIPTION
**Pull request by PVGeorgievPrime**
Replacing AutoRoute with GoRouter as navigation package.

### CODE REVIEW

File: examples/booking_app/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -31,6 +31,16 @@
                 <action android:name="android.intent.action.MAIN"/>
                 <category android:name="android.intent.category.LAUNCHER"/>
             </intent-filter>
+            <!-- Deep links -->
+            <meta-data android:name="flutter_deeplinking_enabled" android:value="true" />
```
Comment (StanevPrime) [inline examples/booking_app/android/app/src/main/AndroidManifest.xml:35]: Please do not forget to add documentation in the README file of the rx_bloc_cli

File: examples/booking_app/ios/Runner/Info.plist [hunk 2]
```diff
@@ -45,5 +45,20 @@
 	<true/>
 	<key>UIApplicationSupportsIndirectInputEvents</key>
 	<true/>
+	<key>FlutterDeepLinkingEnabled</key>
+    <true/>
+    <key>CFBundleURLTypes</key>
```
Comment (StanevPrime) [inline examples/booking_app/ios/Runner/Info.plist:50]: Same as above

File: examples/booking_app/lib/lib_router/blocs/router_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import 'package:go_router/go_router.dart';
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../router.dart';
+
+part 'router_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the NavigationBloC.
+abstract class RouterBlocEvents {
```
Comment (StanevPrime) [inline examples/booking_app/lib/lib_router/blocs/router_bloc.dart:17]: Please do not forget to add documentation here as well as in the mason brick

File: examples/booking_app/lib/lib_router/routes/hotel_details_routes.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
+part of '../router.dart';
```
Comment (StanevPrime) [inline examples/booking_app/lib/lib_router/routes/hotel_details_routes.dart:1]: They are just two routes. Do you think is hood idea to group be into the file routes/routes.dart ?

**Review by StanevPrime — APPROVED** [review]
It looks good. I left a couple of cosmetic comments

File: examples/booking_app/lib/lib_router/routes/hotel_details_routes.dart (hunk 4 above)
Comment (PVGeorgievPrime) [reply examples/booking_app/lib/lib_router/routes/hotel_details_routes.dart]: Done

**Review by StanevPrime — APPROVED** [review]

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**